change/@ni-nimble-angular-e488db99-92c5-4954-a624-673ac1c31f32.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
```

Comment:
I assumed the `template` value was taken into account, too, in some form. Since `AbsoluteSimpleLinkCmp` and `RelativeLinkCmp` have nearly identical templates, and we've only had issues with them. But then again, the templates are not textually identical, and it's less likely Angular is using some parsed/normalized version of that string to determine equality. So maybe lots of these components have potential to collide. I would propose I make all of the selectors unique using `:not()`. That should avoid this for sure. Sound good?

Reply:
Replied in the other comment that we should have a discussion. I don't particularly like the idea that we have to make such big changes to forked tests we need to keep in sync

Reply:
To address the original question, though the `template` string is not directly part of the [ID generation hash](https://github.com/angular/angular/blob/ace0a683f60eba3d30d11f241c59e37386b2b805/packages/core/src/render3/definition.ts#L698), it is indirectly involved via the included [number of bindings](https://github.com/angular/angular/blob/fedeaac8ba85002e34dbd6ffa682d7483d39b5fd/packages/core/src/render3/interfaces/definition.ts#L330), number of nodes, local refs, and pipes, etc. So we were only seeing the issue with those two components, because they were the only ones with the same selector and the same numbers of those other properties.